code/Serving_API.py

```diff
@@ -124,9 +146,62 @@ def calculate_shortest_path_distance(G, shortest_path_cities):
 
         total_distance += G[city1][city2]['weight']
         # Only append the coordinates of the source city here
-        coordinate_list.append(G.nodes[city1]['coord'])
-        wkt_list.append(G[city1][city2]['path_wkt'])
-        cable_type_list.append(G[city1][city2]['cable_type'])
+        # for each edge, find all the available asn nodes that are close to the edge
+        insertable_asn = []
+        for lat, lon in asn_nodes:
+            if lat != 'NULL' and lon != 'NULL':
+                result = find_closest_paths(
```

Comment:
See comments on this function. Since you really just need a boolean flag, there's no need to return the full object.